### DISCUSSION TITLE
Disable useIntelliSenseInDebugging globally

### DISCUSSION THREAD

**Discussion by H-H-C-C** [zero-plusplus/vscode-autohotkey-debug#182] [Q&A] [answered]
How can I disable useIntelliSenseInDebugging globally (without having to create a launch.json file)? I understand that should be doable using "%AppData%\Code\User\settings.json" but I cannot find an example of which lines I should add. I have tried variations of 

```
  "launch": {
    "configurations": [
      {
        "name": "AutoHotkey Launch",
        "type": "autohotkey",
        "request": "launch",
        "port": "9002",
        "variableCategories": "recommend",
        "useDebugDirective": true,
        "useAutoJumpToError": true,
        "useIntelliSenseInDebugging": false
      }
    ],
  },
```

**Comment by zero-plusplus** [comment, marked as answer]
I do not know how to set the default value of the launch.json attribute globally. Perhaps VSCode does not provide such a feature.

Do you disable the IntelliSenseInDebugging feature because you use other extensions that provide IntelliSense?

**Reply by H-H-C-C** [reply]
I use my own extension to provide syntax coloring. My extension does not provide IntelliSense but auto-completion suggestions are achieved through snippets. I would like to disable IntelliSenseInDebugging because it overrides word-based suggestions (see [this post](https://stackoverflow.com/a/55219340)). 

It is my understanding that [there are global settings for launch](https://code.visualstudio.com/docs/getstarted/settings): 

````
  // Global debug launch configuration. Should be used as an alternative to 'launch.json' that is shared across workspaces.
  "launch": {
    "configurations": [],
    "compounds": []
  },
````

**Reply by zero-plusplus** [reply]
> (see this post).

I had looked into this problem before and concluded that the only way to do it was to create word-based completion on the provider side and then merge it. I couldn't understand why such a problem occurs, but now I know the answer. Thank you.

---

> It is my understanding that there are global settings for launch:

You are right, a global debug configuration can be created. Maybe I'm not understanding the intent of your question.

Your first example works correctly in my environment.

Could you please tell me what is not working again?

**Reply by H-H-C-C** [reply]
Since my AHK extension does not provide IntelliSense, it does not override the word-based completion. With vscode-autohotkey-debug disabled, word-based completion works as expected. With vscode-autohotkey-debug enabled, AHK keywords are available for autocompletion but other words (say text or comments) are not longer in the list of suggestions.

I thought that "useIntelliSenseInDebugging:false" meant "do not register the IntelliSense provider at all". I just realized my mistake.

**Reply by zero-plusplus** [reply]
Thank you. I understand now.

You are right, the provider will be used regardless of the attribute setting.

So that's why word-based completion doesn't work this time.

I tried several ways to deal with this problem, but could not solve it.

---

The easiest solution is to create a version that omits the provider registration process.

If you know how to create a vsix, you can use the following method to omit the registration process for the provider.
1. clone vscode-autohotkey-debug
2. run `npm install`.
1. comment out [herer](https://github.com/zero-plusplus/vscode-autohotkey-debug/blob/master/src/extension.ts#L575)
2. remove the import of `completionItemProvider` [here](https://github.com/zero-plusplus/vscode-autohotkey-debug/blob/master/src/extension.ts#L10)
3. install vsce and run `npm run package`
4. Install the vsix you created

**Reply by zero-plusplus** [reply]
In the future, it will be possible to set whether or not to register the provider in settings.json.

**Reply by H-H-C-C** [reply]
Thank you!
